nomos-sdp/src/lib.rs

```diff
@@ -0,0 +1,234 @@
+pub mod state;
+
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+};
+
+use blake2::{Blake2b, Digest};
+use multiaddr::Multiaddr;
+
+pub type StakeThreshold = u64;
+pub type BlockNumber = u64;
```

Comment:
This probably should be `NonZero`.

Reply:
Why couldn't it be zero?

Reply:
Wait, totally brainfart...I was thinking of negative values and mixed everything 😵